app/controllers/interactions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class InteractionsController < ApplicationController
+  def create
+    interaction = build_interaction
+    medium = find_or_create_medium
+    interaction.medium = medium
+    build_notes(interaction)
+
+    if interaction.save
+      redirect_to contact_path(interaction.contact_id), notice: "Interaction was successfully created."
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def build_interaction
+    Interaction.new(contact_id: params[:interaction][:contact_id], interaction_date: params[:interaction][:interaction_date], user: current_user)
+  end
+
+  def find_or_create_medium
+    Medium.find_or_create_by(name: params[:interaction][:medium], user: current_user)
+  end
```

Comment:
Maybe this is crazy, but could you put `current_user` into a variable called `user` and then omit the value?

Reply:
I think both would be good, but let's stick with yours for now!